abi/abi.go

```diff
@@ -87,6 +88,79 @@ func NewABI(s string) (*ABI, error) {
 	return NewABIFromReader(bytes.NewReader([]byte(s)))
 }
 
+func NewABIFromSlice(abi []compiler.AbiField) (*ABI, error) {
```

Comment:
Please add comment to the function, explaining what it does and what are the parameters and return values.